### PR TITLE
Audit report fixing (6.20)

### PR DESCRIPTION
**Pull request by JensonCollins**

### CODE REVIEW

File: contracts/Marketplace.sol [hunk 1]
```diff
@@ -229,7 +229,7 @@ contract Marketplace is ReentrancyGuard, Ownable, Pausable {
     );
   }
 
-  function setWhitelistTokens(address[] memory _whitelist) public onlyOwner {
```
Comment (JensonCollins) [inline contracts/Marketplace.sol:232]: MAR-01S

File: contracts/Marketplace.sol [hunk 2]
```diff
@@ -238,7 +238,7 @@ contract Marketplace is ReentrancyGuard, Ownable, Pausable {
     emit WhitelistAdded(_whitelist);
   }
 
-  function removeWhitelistTokens(address[] memory _whitelist) public onlyOwner {
+  function removeWhitelistTokens(address[] calldata _whitelist) external onlyOwner {
```
Comment (JensonCollins) [inline contracts/Marketplace.sol:241]: MAR-01S

File: contracts/Marketplace.sol [hunk 3]
```diff
@@ -152,7 +153,7 @@ contract Marketplace is ReentrancyGuard, Ownable, Pausable {
     uint256 buyingFee2 = (fee2 * listedPrice / DECIMAL_FACTOR);
 
     if (address(payTokenAddress) == address(0)) {
-      if (msg.value < listedPrice) revert NotEnough();
+      if (msg.value != listedPrice) revert InvalidAmount();
```
Comment (JensonCollins) [inline contracts/Marketplace.sol:156]: MAR-01M

File: contracts/Marketplace.sol [hunk 4]
```diff
@@ -164,6 +165,7 @@ contract Marketplace is ReentrancyGuard, Ownable, Pausable {
       (bool sent, ) = payable(listings[listingId].owner).call{value: listedPrice - buyingFee1 - buyingFee2}("");
       require(sent, "Failed to send Ether");
     } else {
+      if (msg.value != 0) revert InvalidAmount();
```
Comment (JensonCollins) [inline contracts/Marketplace.sol:168]: MAR-02M

File: contracts/BlastEquipmentNFT.sol [hunk 5]
```diff
@@ -63,25 +63,22 @@ contract BlastEquipmentNFT is
         bytes32[] calldata _hash,
         string[] calldata _realUri
     ) external override onlyRole(MINTER_ROLE) {
-        require(_to != address(0));
+        require(_to != address(0), "To address can't be zero");
```
Comment (JensonCollins) [inline contracts/BlastEquipmentNFT.sol:66]: BEN-01C

File: contracts/BlastEquipmentNFT.sol [hunk 6]
```diff
@@ -63,25 +63,22 @@ contract BlastEquipmentNFT is
...
-            _setTokenURI(tokenId, _uri[i]);
-            emit AttributeAdded(tokenId, 1, 0, 0, 0);
+            _safeMint(_to, _uri[i], _hash[i], _realUri[i]);
         }
     }
 
-    function safeMintReplicator(address _to, string calldata _uri, bytes32 _hash, string calldata _realUri) external override onlyRole(REPLICATOR_ROLE) returns (uint) {
-        require(_to != address(0));
+    function safeMintReplicator(address _to, string calldata _uri, bytes32 _hash, string calldata _realUri) external override onlyRole(REPLICATOR_ROLE) returns (uint256) {
+        require(_to != address(0), "To address can't be zero");
```
Comment (JensonCollins) [inline contracts/BlastEquipmentNFT.sol:76]: BEN-01C

File: contracts/BlastEquipmentNFT.sol [hunk 7]
```diff
@@ -63,25 +63,22 @@ contract BlastEquipmentNFT is
...
 
+        return _safeMint(_to, _uri, _hash, _realUri);
+    }
+
+    function _safeMint(address _to, string memory _uri, bytes32 _hash, string memory _realUri) internal returns (uint256) {
```
Comment (JensonCollins) [inline contracts/BlastEquipmentNFT.sol:81]: BEN-02C

File: contracts/BlastLootBox.sol [hunk 8]
```diff
@@ -58,7 +58,7 @@ contract BlastLootBox is
         onlyRole(DEFAULT_ADMIN_ROLE)
     {
         if (_to.length != _uri.length || _to.length != _eqtIds.length) revert InvalidParams();
-        if (!(_tokenType == 1 || _tokenType == 2)) revert InvalidParams();
+        if (_tokenType != 1 && _tokenType != 2) revert InvalidParams();
```
Comment (JensonCollins) [inline contracts/BlastLootBox.sol:61]: BLB-01C

File: contracts/BlastLootBox.sol [hunk 9]
```diff
@@ -28,16 +28,16 @@ contract BlastLootBox is
...
+        uint256 token1;
+        uint256 token2;
     }
 
     bytes32 public constant GAME_ROLE = keccak256("GAME_ROLE");
 
     Counters.Counter public _tokenIdCounter;
-    mapping(uint => LootBox) private lootboxDetails;
-    mapping(uint => uint8) private tokenTypes;
+    mapping(uint256 => LootBox) private lootboxDetails;
```
Comment (JensonCollins) [inline contracts/BlastLootBox.sol:39]: BLB-02C

File: contracts/Marketplace.sol [hunk 10]
```diff
@@ -118,24 +118,25 @@ contract Marketplace is ReentrancyGuard, Ownable, Pausable {
...
+    listing.isActive = false;
     erc721Contract.transferFrom(
       address(this),
       _msgSender(),
-      listings[listingId].tokenId
+      listing.tokenId
     );
     activeListingCount = activeListingCount - 1;
-    emit ItemDelisted(listingId, listings[listingId].tokenId, _msgSender() );   
+    emit ItemDelisted(listingId, listing.tokenId, _msgSender() );
```
Comment (JensonCollins) [inline contracts/Marketplace.sol:139]: MAR-01C

File: contracts/Marketplace.sol [hunk 11]
```diff
@@ -209,8 +211,6 @@ contract Marketplace is ReentrancyGuard, Ownable, Pausable {
   /// @param _treasury2 New treasury2 address.
   function setFee(uint256 _fee1, address _treasury1, uint256 _fee2, address _treasury2) public onlyOwner
   {
-    if (_fee1 >= DECIMAL_FACTOR) revert InvalidParam();
-    if (_fee2 >= DECIMAL_FACTOR) revert InvalidParam();
```
Comment (JensonCollins) [inline contracts/Marketplace.sol:213]: MAR-02C

**Comment by mrfungfung** [comment]
Looks good to me

**Pull request merged**